py3status/modules/diskdata.py

```diff
@@ -130,6 +135,9 @@ def _get_diskstats(self, disk):
         if disk and disk.startswith("/dev/"):
             disk = disk[5:]
 
+        if not os.path.exists("/proc/diskstats"):
+            return read, write
+
```

Comment:
No `/proc/diskstats`? OK, we can skip `/proc/diskstats`-related placeholders instead of faking it.
```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index 1fd9933..9b2cefa 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -90,8 +90,12 @@ class Py3status:
                 self.init[name] = {"placeholders": placeholders, "keys": match}
 
         if self.init["diskstats"]:
-            self.last_diskstats = self._get_diskstats(self.disk)
-            self.last_time = time.monotonic()
+            try:
+                self.last_diskstats = self._get_diskstats(self.disk)
+                self.last_time = time.monotonic()
+            except Exception:
+                self.init["diskstats"] = []
+
 
         self.thresholds_init = self.py3.get_color_names_list(self.format)
 

```

Reply:
Note: Any other exceptions (ie invalid /dev/disk) could turn diskstats off too.

Reply:
Not returning numbers is much better indeed. This however still shows the  {total} placeholder. What's the best way to deal with this? Just changing config to omit it? I wonder if it could be left out implictily but that would probably also break users expectations

Reply:
Yup. Changing `format` config to omit that placeholder locally is the way...or to hide it by default... like this.

```diff
diff --git a/py3status/modules/diskdata.py b/py3status/modules/diskdata.py
index 1fd9933..c48268f 100644
--- a/py3status/modules/diskdata.py
+++ b/py3status/modules/diskdata.py
@@ -65,7 +65,7 @@ class Py3status:
     # available configuration parameters
     cache_timeout = 10
     disk = None
-    format = "{disk}: {used_percent}% ({total})"
+    format = "{disk}: {used_percent}%[ ({total})]"
     format_rate = r"[\?min_length=11 {value:.1f} {unit}]"
     format_space = r"[\?min_length=5 {value:.1f}]"
     sector_size = 512
```

Reply:
Cool, thanks! Patches updated